### ISSUE TITLE
optimization_level=1 always uses Trivial layout

### ISSUE THREAD

**Issue by nonhermitian** [Qiskit/qiskit-terra#2845]
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
```
               ┌───┐                ░ ┌─┐            
q37_0: |0>─────┤ X ├────────────────░─┤M├────────────
          ┌───┐└─┬─┘┌───┐           ░ └╥┘┌─┐         
q37_1: |0>┤ H ├──■──┤ X ├───────────░──╫─┤M├─────────
          ├───┤     └─┬─┘┌───┐      ░  ║ └╥┘┌─┐      
q37_2: |0>┤ H ├───────■──┤ X ├──────░──╫──╫─┤M├──────
          ├───┤          └─┬─┘┌───┐ ░  ║  ║ └╥┘┌─┐   
q37_3: |0>┤ H ├────────────■──┤ X ├─░──╫──╫──╫─┤M├───
          ├───┤               └─┬─┘ ░  ║  ║  ║ └╥┘┌─┐
q37_4: |0>┤ H ├─────────────────■───░──╫──╫──╫──╫─┤M├
          └───┘                     ░  ║  ║  ║  ║ └╥┘
  c5_0: 0 ═════════════════════════════╩══╬══╬══╬══╬═
                                          ║  ║  ║  ║ 
  c5_1: 0 ════════════════════════════════╩══╬══╬══╬═
                                             ║  ║  ║ 
  c5_2: 0 ═══════════════════════════════════╩══╬══╬═
                                                ║  ║ 
  c5_3: 0 ══════════════════════════════════════╩══╬═
                                                   ║ 
  c5_4: 0 ═════════════════════════════════════════╩═
```
```python
backend = provider.get_backend('ibmq_20_tokyo')
new_qc = transpile(qc, backend, optimization_level=1)
```

shows initial layout as the trivial layout, which is disconnected:

<img width="296" alt="Screen Shot 2019-07-22 at 4 14 54 PM" src="https://user-images.githubusercontent.com/1249193/61662177-e564af00-ac9b-11e9-94bb-0dbe3fbe0e20.png">

However part #2 in the source says `_layout_check` does better layout 
if the circuit needs swaps, 

```python
# 2. Use a better layout on densely connected qubits, if circuit needs swaps
```
which is obviously the case here, but no other layouts
are actually selected (or imported).

### What is the expected behavior?
Expected that the `DenseLayout` is used if there are swaps needed.

**Comment by 1ucian0** [comment]
Confirmed.
```python
qr = QuantumRegister(5)
cr = ClassicalRegister(5)
qc = QuantumCircuit(qr, cr)

qc.h(qr[1])
qc.h(qr[2])
qc.h(qr[3])
qc.h(qr[4])
qc.cx(qr[0], qr[1])
qc.cx(qr[1], qr[2])
qc.cx(qr[2], qr[3])
qc.cx(qr[3], qr[4])
qc.barrier(qr)
qc.measure(qr, cr)

backend = IBMQ.get_backend('ibmq_20_tokyo')
new_qc = transpile(qc, backend, optimization_level=1)
print('\n'.join([str((i, j)) for i, j in new_qc.layout._p2v.values() if i.name != 'ancilla']))
```
```
(QuantumRegister(5, 'q0'), 0)
(QuantumRegister(5, 'q0'), 1)
(QuantumRegister(5, 'q0'), 2)
(QuantumRegister(5, 'q0'), 3)
(QuantumRegister(5, 'q0'), 4)
```

Is also `optimization_level=2` having a similar issue?
```python
new_qc = transpile(qc, backend, optimization_level=2)
print('\n'.join([str((i, j)) for i, j in new_qc.layout._p2v.values() if i.name != 'ancilla']))
```
```
(QuantumRegister(5, 'q0'), 1)
(QuantumRegister(5, 'q0'), 4)
(QuantumRegister(5, 'q0'), 2)
(QuantumRegister(5, 'q0'), 3)
(QuantumRegister(5, 'q0'), 0)
```

**Comment by kdk** [comment]
Re-opened by #2975, will revisit post-0.9.

**Comment by 1ucian0** [comment]
This can be fixed once "best of" approach is implemented (see #2969). On hold until then.

**Comment by 1ucian0** [comment]
Under the assumption that `DenseLayout` is always better than `TrivialLayout` (is that true?), the PR https://github.com/Qiskit/qiskit-terra/pull/3657 is a simpler approach (if `TriviaLayout` does not return a perfect scoring, then it uses `DenseLayout`).
